oak-blob-cloud/src/test/java/org/apache/jackrabbit/oak/blob/cloud/s3/TestS3DSWithSSEKMS.java

```diff
@@ -0,0 +1,108 @@
+import static org.junit.Assert.fail;
+
+/**
+ * Test S3DataStore operation with SSE_S3 encryption.
+ * It requires to pass aws config file via system property  or system properties by prefixing with 'ds.'.
+ * See details @ {@link S3DataStoreUtils}.
+ * For e.g. -Dconfig=/opt/cq/aws.properties. Sample aws properties located at
+ * src/test/resources/aws.properties
+
+ */
+public class TestS3DSWithSSEKMS extends TestS3Ds {
+
+        protected static final Logger LOG = LoggerFactory.getLogger(TestS3DSWithSSES3.class);
+
+        protected static long ONE_KB = 1024;
+        protected static long ONE_MB = ONE_KB * ONE_KB;
+        protected static long TEN_MB = ONE_MB * 10;
+        protected static long ONE_HUNDRED_MB = ONE_MB * 100;
+        protected static long ONE_GB = ONE_HUNDRED_MB * 10;
+        
+        @Override
+        @Before
+        public void setUp() throws Exception {
+            super.setUp();
+            props = getS3Config();
+            props.setProperty(S3Constants.S3_ENCRYPTION, S3Constants.S3_ENCRYPTION_SSE_KMS);
+            props.setProperty("cacheSize", "0");
+        }
+
+        /**
+         * Test data migration enabling SSE_KMS encryption.
+         */
+        @Test
+        public void testDataMigration() {
+            try {
+                //manually close the setup ds and remove encryption
+                ds.close();
+                props.remove(S3Constants.S3_ENCRYPTION);
+                ds = createDataStore();
+
+                byte[] data = new byte[dataLength];
+                randomGen.nextBytes(data);
+                DataRecord rec = ds.addRecord(new ByteArrayInputStream(data));
+                Assert.assertEquals(data.length, rec.getLength());
+                assertRecord(data, rec);
+                ds.close();
+
+                // turn encryption now anc recreate datastore instance
+                props.setProperty(S3Constants.S3_ENCRYPTION, S3Constants.S3_ENCRYPTION_SSE_KMS);
+                props.setProperty(S3Constants.S3_RENAME_KEYS, "true");
+                ds = createDataStore();
+
+                rec = ds.getRecord(rec.getIdentifier());
+                Assert.assertEquals(data.length, rec.getLength());
+                assertRecord(data, rec);
+
+                randomGen.nextBytes(data);
+                ds.addRecord(new ByteArrayInputStream(data));
+                ds.close();
+            } catch (Exception e) {
+                LOG.error("error:", e);
+                fail(e.getMessage());
+            }
+        }
+
+        @Test
+        public void testInitiateDirectUploadUnlimitedURIs() throws DataRecordUploadException {
+            ConfigurableDataRecordAccessProvider ds = null;
+            try {
+                ds = (ConfigurableDataRecordAccessProvider) createDataStore();
+            } catch (RepositoryException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
```

Comment:
Let's not catch the exception here . Just throw it - that would make the test fail at this step only.